File: sol/solution/0001-0100/0005. Longest Palindromic Substring/Solution.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        n = len(s)
        start = max_len = 0
        for i in range(n):
            left = right = i
            while left >= 0 and right < n and s[left] == s[right]:
                if right - left + 1 > max_len:
                    start = left
                    max_len = right - left + 1
                left -= 1
                right += 1
            left = i
            right = i + 1
            while left >= 0 and right < n and s[left] == s[right]:
                if right - left + 1 > max_len:
                    start = left
                    max_len = right - left + 1
                left -= 1
                right += 1
        return s[start:start + max_len]

File: sol/solution/0001-0100/0005. Longest Palindromic Substring/Solution.kt
class Solution {
    fun longestPalindrome(s: String): String {
        val n = s.length
        var start = 0
        var maxLen = 1
        for (i in 0 until n) {
            var left = i
            var right = i
            while (left >= 0 && right < n && s[left] == s[right]) {
                if (right - left + 1 > maxLen) {
                    start = left
                    maxLen = right - left + 1
                }
                left--
                right++
            }
            left = i
            right = i + 1
            while (left >= 0 && right < n && s[left] == s[right]) {
                if (right - left + 1 > maxLen) {
                    start = left
                    maxLen = right - left + 1
                }
                left--
                right++
            }
        }
        return s.substring(start, start + maxLen)
    }
}